elements/pf-popover/pf-popover.ts

```diff
@@ -0,0 +1,389 @@
+ *          Header bottom margin
+ *          {@default `0.5rem`}
+ * @cssprop {<number>} --pf-c-popover__title--LineHeight
+ *          Header line height
+ *          {@default `1.5`}
+ * @cssprop {<string>} --pf-c-popover__title--FontFamily
+ *          Header font-family
+ *          {@default `'RedHatDisplay', 'Overpass', overpass, helvetica, arial, sans-serif`}
+ * @cssprop {<length>} --pf-c-popover__footer--MarginTop
+ *          Footer top margin
+ *          {@default `1rem`}
+ * @cssprop {<color>} --pf-c-popover--m-default__title-text--Color
+ *          Default alert heading color
+ *          {@default `#003737`}
+ * @cssprop {<color>} --pf-c-popover--m-default__title-icon--Color
+ *          Default alert icon color
+ *          {@default `#009596`}
+ * @cssprop {<color>} --pf-c-popover--m-info__title-text--Color
+ *          Default alert heading color
+ *          {@default `#002952`}
+ * @cssprop {<color>} --pf-c-popover--m-info__title-icon--Color
+ *          Default alert icon color
+ *          {@default `#2b9af3`}
+ * @cssprop {<color>} --pf-c-popover--m-warning__title-text--Color
+ *          Default alert heading color
+ *          {@default `#795600`}
+ * @cssprop {<color>} --pf-c-popover--m-warning__title-icon--Color
+ *          Default alert icon color
+ *          {@default `#f0ab00`}
+ * @cssprop {<color>} --pf-c-popover--m-success__title-text--Color
+ *          Default alert heading color
+ *          {@default `#1e4f18`}
+ * @cssprop {<color>} --pf-c-popover--m-success__title-icon--Color
+ *          Default alert icon color
+ *          {@default `#3e8635`}
+ * @cssprop {<color>} --pf-c-popover--m-danger__title-text--Color
+ *          Default alert heading color
+ *          {@default `#a30000`}
+ * @cssprop {<color>} --pf-c-popover--m-danger__title-icon--Color
+ *          Default alert icon color
+ *          {@default `#c9190b`}
+ */
+@customElement('pf-popover')
+export class PfPopover extends LitElement {
+  static readonly styles = [styles];
+
+  // todo: alertSeverityScreenReaderText
+  // todo: hideOnOutsideClick
+  // todo: withFocusTrap
+  @property({ reflect: true }) position: Placement = 'top';
+  @property({ reflect: true }) heading?: string;
+  @property({ reflect: true }) body = '';
+  @property({ reflect: true }) footer?: string;
+  @property({ reflect: true }) icon?: string;
+  @property({ reflect: true }) label?: string;
+  @property({ type: Number, reflect: true }) distance?: number = 25;
+  // todo: handle PF4s 'flip' Placement option in flip-behavior if possible
+  // https://www.patternfly.org/v4/components/popover#popover
+  @property({ attribute: 'flip-behavior', converter: StringListConverter }) flipBehavior?: Placement[] = ['top', 'bottom', 'left', 'right', 'top-start', 'top-end', 'bottom-start', 'bottom-end', 'left-start', 'left-end', 'right-start', 'right-end'];
+  @property({ type: Boolean, reflect: true, attribute: 'enable-flip' }) enableFlip?: boolean = true;
```

Comment:
```suggestion
  @property({ type: Boolean, reflect: true, attribute: 'enable-flip' }) enableFlip?: boolean = true;
```

a reflecting boolean attr that defaults to true isn't great because there will be no good way to unset. maybe flip (zing!) the sense to `no-flip`?